Basic/2DSpaceShooter/Assets/Scripts/RandomPositionPlayerSpawner.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using MLAPI;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+[RequireComponent(typeof(NetworkManager))]
+public class RandomPositionPlayerSpawner: MonoBehaviour
+{
+    NetworkManager m_NetworkManager;
+
+    int m_RoundRobinIndex = 0;
+    
+    [SerializeField]
+    SpawnMethod m_SpawnMethod;
+    
+    [SerializeField]
+    List<Vector3> m_SpawnPositions = new List<Vector3>() { Vector3.zero };
+
+    /// <summary>
+    /// Get a spawn position for a spawned object based on the spawn method.
+    /// </summary>
+    /// <returns>?The spawn position.</returns>
+    /// <exception cref="NotImplementedException"></exception>
+    public Vector3 GetNextSpawnPosition()
+    {
+        switch (m_SpawnMethod)
+        {
+            case SpawnMethod.Random:
+                var index = Random.Range(0, m_SpawnPositions.Count);
+                return m_SpawnPositions[index];
+            case SpawnMethod.RoundRobin:
+                m_RoundRobinIndex++;
```

Comment:
This should be reset once it reaches max available spawn points. we could just use  m_RoundRobinIndex = (m_RoundRobinIndex + 1) % m_SpawnPositions.Count;

Reply:
Good catch, my plan of intent was to use the modulus here but forgot it